homeassistant/components/switch/flux.py

```diff
@@ -172,30 +174,30 @@ def is_on(self):
         """Return true if switch is on."""
         return self.unsub_tracker is not None
 
-    def turn_on(self, **kwargs):
+    async def async_turn_on(self, **kwargs):
         """Turn on flux."""
         if self.is_on:
             return
 
         # Make initial update
-        self.flux_update()
+        await self.async_flux_update()
```

Comment:
You have to move the `self.unsub_tracker` assignment above this `await` if you want to remove the race.